phpseclib/Net/SSH2.php

```diff
@@ -1499,16 +1499,16 @@ private function key_exchange($kexinit_payload_server = false): bool
 
         $exchange_hash_rfc4419 = '';
 
-        if (strpos($this->kex_algorithm, 'curve25519-sha256') === 0 || strpos($this->kex_algorithm, 'ecdh-sha2-nistp') === 0) {
-            $curve = strpos($this->kex_algorithm, 'curve25519-sha256') === 0 ?
+        if (str_starts_with($this->kex_algorithm, 'curve25519-sha256')   || str_starts_with($this->kex_algorithm, 'ecdh-sha2-nistp')) {
+            $curve = str_starts_with($this->kex_algorithm, 'curve25519-sha256')   ?
```

Comment:
Looks like there are some double spaces here that ought to be reduced to a single space